### ISSUE TITLE
Properties of BaseSchemaObject

### ISSUE THREAD

**Issue by HerrRomich** [kogosoftwarellc/open-api#312]
According to JSON Schema specification elements of Object "properties" could be either Schema Objects 
 or References. This means for me that in "[open-api/packages/openapi-types/index.ts](https://github.com/kogosoftwarellc/open-api/blob/master/packages/openapi-types/index.ts)" instead of Schema Object:
```typescript
  interface BaseSchemaObject {
    // JSON schema allowed properties, adjusted for OpenAPI
    title?: string;
...
    properties?: {
      [name: string]: SchemaObject;
    };
...  
    example?: any;
    deprecated?: boolean;
  }
```

should be a union:
``` typescript
  interface BaseSchemaObject {
    // JSON schema allowed properties, adjusted for OpenAPI
    title?: string;
...
    properties?: {
      [name: string]: ReferenceObject | SchemaObject;
    };
...  
    example?: any;
    deprecated?: boolean;
  }
```
Thank you

**Comment by HerrRomich** [comment]
Should I have a permission to create a branch?

**Comment by jsdevel** [comment]
of course!

**Comment by jsdevel** [comment]
oh, you can fork this repo and open a PR.

**Comment by jsdevel** [comment]
closed by #313
